_posts/2018/2018-08-19-use-the-presence-method.md

```diff
@@ -0,0 +1,80 @@
+---
+title: Use the presence method
+description: 'An unsung gem from Active Support'
+layout: article
+category: ruby
+image:
+  base: '2018/use-the-presence-method'
+  alt: 'Ghostly'
+  source: 'https://unsplash.com/photos/ZC0EbdLC8G0'
+  credit: 'Stefano Pollio'
+---
+
+Active Support takes a lot flak for the magic it adds to core Ruby libraries including patching a lot of functionality into Ruby's `Object`.
+
+The documentation can be found [here](https://api.rubyonrails.org/classes/Object.html#method-i-presence).
+
+For me, a lot of the delight in Rails comes from the shortcuts it adds. Some features even eventually make their way into Ruby core.
+
+
+## Instead of…
+
+...using long-winded conditionals to display the value of a variable.
+
+```ruby
+class User < ApplicationRecord
+  validates :email, presence: true
+
+  def full_name
+    if nickname.present?
+      nickname
+    elsif given_name.present?
+      given_name
+    else
+      email.split('@').first
```

Comment:
Not exactly.

```ruby
'nodunayo@email.com'.split('@').first
# -> 'nodunayo'
```

plus it's a fallback :-)

Reply:
But ‘nickname’ and ‘given name’ are all examples of the first name, right? I’m saying, even with your fallback, a more suitable name for this method seems to be ‘first_name’, rather than ‘full_name’.